### PR TITLE
DAG: Fix chain mismanagement in SoftenFloatRes_FP_EXTEND

### PR DESCRIPTION
**Pull request by arsenm**
This would result in nodes not getting appropriately re-legalized
in the strictfp case.

### CODE REVIEW

File: llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp [hunk 1]
```diff
@@ -2181,6 +2186,20 @@ static ISD::NodeType GetPromotionOpcode(EVT OpVT, EVT RetVT) {
...
 }
 
+static ISD::NodeType GetPromotionOpcodeStrict(EVT OpVT, EVT RetVT) {
+  if (OpVT == MVT::f16) {
+    return ISD::STRICT_FP16_TO_FP;
+  } else if (RetVT == MVT::f16) {
+    return ISD::STRICT_FP_TO_FP16;
+  } else if (OpVT == MVT::bf16) {
+    // return ISD::STRICT_BF16_TO_FP;
+  } else if (RetVT == MVT::bf16) {
```
Comment (RKSimon) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp:2196]: (style) Break if-else chain

File: llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp [hunk 2]
```diff
@@ -2276,6 +2298,26 @@ SDValue DAGTypeLegalizer::PromoteFloatOp_FP_EXTEND(SDNode *N, unsigned OpNo) {
   return DAG.getNode(ISD::FP_EXTEND, SDLoc(N), VT, Op);
 }
 
+SDValue DAGTypeLegalizer::PromoteFloatOp_STRICT_FP_EXTEND(SDNode *N,
+                                                          unsigned OpNo) {
+  assert(OpNo == 1);
```
Comment (RKSimon) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp:2303]: (style) assertion message

File: llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp [hunk 3]
```diff
@@ -1804,6 +1816,7 @@ bool DAGTypeLegalizer::PromoteIntegerOperand(SDNode *N, unsigned OpNo) {
   case ISD::FP16_TO_FP:
   case ISD::VP_UINT_TO_FP:
   case ISD::UINT_TO_FP:   Res = PromoteIntOp_UINT_TO_FP(N); break;
+  case ISD::STRICT_FP16_TO_FP:
```
Comment (RKSimon) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp:1819]: This just doesn't look goo imo. Why not add PromoteIntOp_STRICT_FP16_TO_FP? Or rename PromoteIntOp_STRICT_UINT_TO_FP?

**Review by RKSimon — COMMENTED** [review]
A few minor observations

File: llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp (hunk 3 above)
Comment (arsenm) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp:1819]: The non-strict path does the same thing, so I'm just copying that. I don't see much point in renaming it

File: llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp (hunk 2 above)
Comment (arsenm) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp]: This is also from the parent commit, this PR is supposed to just be the top one

**Comment by arsenm** [comment]
Messed up the PR by accidentally pushing commits and also addressed comments in wrong PR

**Pull request closed without merging**